src/types/io.ts

```diff
@@ -312,6 +315,14 @@ export interface AoIORead {
   }: {
     address: WalletAddress;
   }): Promise<AoGateway | undefined>;
+  // TODO: these could be moved to a separate Gateways class that implements gateway specific interactions
+  getGatewayDelegates(
+    params?: PaginationParams,
```

Comment:
The input type here is different the actual implementation (where `address` is unioned into the type). Same below.